docs/scripts_md/ImagingUpload.md

```diff
@@ -0,0 +1,163 @@
+  - $pname               : patient name
+  - $profile             : name of the configuration file
+                          (typically `prod`)
+
+RETURNS: new instance of this class
+
+### IsCandidateInfoValid()
+
+Validates the File to be uploaded. If the validation passes, the following
+actions will happen:
+  1) Copy the file from tmp folder to /data/incoming
+  2) Set `IsCandidateInfoValidated` to TRUE in the mri\_upload table
+
+RETURNS: 1 on success, 0 on failure
+
+### runDicomTar()
+
+This method executes the following actions:
+ - Runs `dicomTar.pl` with `-clobber -database -profile prod` options
+ - Extracts the TarchiveID of the tarchive created by `dicomTar.pl`
+ - Updates the mri\_upload table if `dicomTar.pl` ran successfully
+
+RETURNS: 1 on success, 0 on failure
+
+### getTarchiveFileLocation()
+
+This method fetches the location of the archive from the tarchive table of
+the database.
+
+RETURNS: the archive location
+
+### runTarchiveLoader()
+
+This methods will call `tarchiveLoader` with the `-clobber -profile prod`
+options and update the mri\_upload table accordingly if `tarchiveLoader` ran
+successfully.
+
+RETURNS: 1 on success, 0 on failure
+
+### PatientNameMatch($dicom\_file)
+
+This method extracts the patient name field from the DICOM file header using
+`dcmdump` and compares it with the patient name information stored in the
+mri\_upload table.
+
+INPUT: full path to the dicom-file
+
+RETURNS: 1 on success, 0 on failure
+
+### isDicom($dicom\_file)
+
+This method checks whether the file given as an argument is of type DICOM.
+
+INPUT: full path to the dicom-file
+
+RETURNS: 1 if file is of type DICOM, 0 if file is not of type DICOM
+
+### runCommandWithExitCode($command)
+
+This method will run any linux command given as an argument using the
+`system()` method and will return the proper exit code.
+
+INPUT: the linux command to be executed
+
+RETURNS: the exit code of the command
+
+### runCommand($command)
+
+This method will run any linux command given as an argument using back-tilt
+and will return the back-tilt return value (which is STDOUT).
+
+INPUT: the linux command to be executed
+
+RETURNS: back-tilt return value (STDOUT)
+
+### CleanUpDataIncomingDir($uploaded\_file)
+
+This method cleans up and removes the uploaded file from the data directory
+once the uploaded file has been inserted into the database and saved in the
+tarchive folder.
+
+RETURNS: 1 on success, 0 on failure
+
+### spool($message, $error, $verb)
+
+This method calls the `Notify->spool` function to log all messages
+returned by the insertion scripts.
+
+INPUT:
+ - $message: message to be logged in the database
+ - $error  : 'Y' for an error log , 'N' otherwise
+ - $verb   : 'N' for few main messages,
+             'Y' for more messages (for developers)
+
+### updateMRIUploadTable($field, $value)
+
+This method updates the `mri_upload` table with `$value` for the field
+`$field`.
+
+INPUT:
```

Comment:
INPUTS (same comment as above, but am not sure if we decided to make it plural form if we have more than one INPUT)